go_modules/spec/dependabot/go_modules/update_checker_spec.rb

```diff
@@ -75,4 +79,58 @@ module foobar
     it "doesn't updates Git SHAs to releases that don't include them"
     it "doesn't update Git SHAs not on master to newer commits to master"
   end
+
+  describe "#lowest_security_fix_version" do
+    subject(:lowest_security_fix_version) { checker.lowest_security_fix_version }
+
+    let(:dependency_version) { "1.0.1" }
+
+    let(:security_advisories) do
+      [
+        Dependabot::SecurityAdvisory.new(
+          dependency_name: dependency_name,
+          package_manager: "go_modules",
+          vulnerable_versions: ["= 1.0.1"]
+        )
+      ]
+    end
+
+    context "when a supported newer version is available" do
+      it "updates to the least new supported version" do
+        is_expected.to eq(Dependabot::GoModules::Version.new("1.0.5"))
+      end
+    end
+  end
+
+  describe "#lowest_resolvable_security_fix_version" do
+    subject(:lowest_resolvable_security_fix_version) { checker.lowest_resolvable_security_fix_version }
+
+    let(:dependency_version) { "1.0.1" }
```

Comment:
Is there a test case that could be added to cover the `unless dependency.top_level?` code-path?

Reply:
Looks like this will be an `indirect` dependency?

Reply:
```
context "with an indirect dependency"
  let(:requirements) { [] }

  it "updates to the least new supported version" do
      is_expected.to eq(Dependabot::GoModules::Version.new(dependency_version))
  end
end
```